packages/trading/src/validation.ts

```diff
@@ -0,0 +1,293 @@
+  }
+
+  if (!params.appCode || params.appCode.trim().length === 0) {
+    throw new TradeValidationError('App code is required and cannot be empty', 'appCode', params.appCode)
+  }
+
+  if (!isValidAmount(params.amount)) {
+    throw new TradeValidationError(
+      `Invalid amount: ${params.amount}. Must be a positive integer in wei`,
+      'amount',
+      params.amount,
+    )
+  }
+
+  validateCommonFields(params)
+}
+
+/**
+ * Validates base trade parameters (without chain ID)
+ */
+export function validateTradeParameters(params: TradeParameters): void {
+  if (!isValidAmount(params.amount)) {
+    throw new TradeValidationError(
+      `Invalid amount: ${params.amount}. Must be a positive integer in wei`,
+      'amount',
+      params.amount,
+    )
+  }
+
+  validateCommonFields(params)
+}
+
+/**
+ * Validates limit trade parameters
+ */
+export function validateLimitTradeParameters(params: LimitTradeParameters): void {
+  if (!isValidAmount(params.sellAmount)) {
+    throw new TradeValidationError(
+      `Invalid sell amount: ${params.sellAmount}. Must be a positive integer in wei`,
+      'sellAmount',
+      params.sellAmount,
+    )
+  }
+
+  if (!isValidAmount(params.buyAmount)) {
+    throw new TradeValidationError(
+      `Invalid buy amount: ${params.buyAmount}. Must be a positive integer in wei`,
+      'buyAmount',
+      params.buyAmount,
+    )
+  }
+
+  if (params.quoteId !== undefined && params.quoteId !== null) {
+    if (typeof params.quoteId === 'number' && (params.quoteId < 0 || !Number.isInteger(params.quoteId))) {
+      throw new TradeValidationError(
+        `Invalid quote ID: ${params.quoteId}. Must be null, undefined, or a positive integer`,
+        'quoteId',
+        params.quoteId,
+      )
+    }
+  }
+
+  validateCommonFields(params)
+}
+
+/**
+ * Validates parameters with detailed error reporting - returns results instead of throwing
+ */
+export function validateSwapParametersSafe(params: SwapParameters): {
+  isValid: boolean
+  errors: TradeValidationError[]
+} {
+  const errors: TradeValidationError[] = []
+
+  try {
+    validateSwapParameters(params)
+  } catch (error) {
+    if (error instanceof TradeValidationError) {
+      errors.push(error)
+    }
+  }
+
+  return {
+    isValid: errors.length === 0,
+    errors,
+  }
+}
+
+/**
+ * Creates a user-friendly validation summary
+ */
+export function getValidationSummary(errors: TradeValidationError[]): string {
+  if (errors.length === 0) return 'All parameters are valid'
+
+  const summary = errors.map((error, index) => `${index + 1}. ${error.field}: ${error.message}`).join('\n')
+
+  return `Found ${errors.length} validation error${errors.length > 1 ? 's' : ''}:\n${summary}`
+}
+
+/**
+ * Validates common edge cases that can cause silent failures
+ */
+export function validateTradeAmounts(sellAmount: string, buyAmount: string, slippageBps?: number): void {
```

Comment:
I don't think `validateTradeAmounts` is correct, we acrually allow amounts less than 1000 wei